src/csharp/CSharpParser.g4

```diff
@@ -178,7 +178,6 @@ unary_expression
 	| '--' unary_expression
 	| OPEN_PARENS type CLOSE_PARENS unary_expression
 	| AWAIT unary_expression
-	| THROW unary_expression
```

Comment:
Why is it unnecessary? Aren't you adding it?

Reply:
Because you can throw more than just unary expressions. In the test case I'm throwing the result of an overloaded `+`. I added throw expressions under the non assignment expression node instead.